### PR TITLE
[Embed API] Python Wrapper

### PR DESCRIPTION
**Pull request by charleslai**
Mostly done, waiting for review

### CODE REVIEW

**Comment by charleslai** [comment]
haha actually not done yet - was leaving a pr open for comments but it seems like i should wait till i have finished everything first

**Comment by talwai** [comment]
Cool thanks, adding a test for the new functionality [here](https://github.com/DataDog/datadogpy/blob/master/tests/integration/api/test_api.py) would also be much appreciated

**Comment by yannmh** [comment]
@charleslai is it ready for review ?

**Comment by charleslai** [comment]
I think I'm good! let me know what needs to be fixed

**Comment by yannmh** [comment]
Thanks a lot @charleslai !

Could you clean/concatenate your git commit history to have something short and clear ? Also could you remove the `merge master` commits ? it's making the PR heavier, and the history confusing (prefer a rebase instead) :)

**Comment by yannmh** [comment]
Do we plan to bring this new feature to dogshell (aka `dog` command) ?

**Comment by charleslai** [comment]
not at the moment I think

**Comment by charleslai** [comment]
cleaned git history - add api wrapper and tests

**Comment by yannmh** [comment]
Great, I'll review your PR soon. Could you also update the [documentation](https://github.com/DataDog/documentation) to reflect your changes please ?

**Comment by charleslai** [comment]
Documentation is updated

File: datadog/api/graphs.py [hunk 1]
```diff
@@ -45,3 +50,91 @@ def status(cls, snapshot_url):
...
+
+    @classmethod
+    def get_all(cls):
+        """
+        Returns a JSON object containing a list of all embeddable graphs
+        in the API user's organization.
+
+        :returns: JSON response from HTTP API request
+        """
+        return super(Embed, cls).get_all()
```
Comment (yannmh) [inline datadog/api/graphs.py]: Any reason why you redefine the method when no modification is made ?

File: datadog/api/graphs.py [hunk 2]
```diff
@@ -45,3 +50,91 @@ def status(cls, snapshot_url):
...
+        """
+        Enable a specified embed.
+
+        :param embed_id: embed token
+        :type embed_id: string embed token
+
+        :returns: JSON response from HTTP API request
+        """
+        handle = embed_id + "/enable"
+        return super(Embed, cls).get(handle)
```
Comment (yannmh) [inline datadog/api/graphs.py]: Can we modify the `ActionAPIResource` class and `_trigger_action` method to comply with this type of handle please ? That would make more sense I think.

File: tests/integration/api/test_api.py [hunk 3]
```diff
@@ -753,5 +753,152 @@ def test_host_muting(self):
...
+        size = "medium"
+        legend = "no"
+        graph_json = json.dumps(graph_def)
+        created_graph = dog.Embed.create(graph_json=graph_json, timeframe=timeframe, size=size, legend=legend)
+        # Save the embed_id into a variable and enable it again
+        embed_id = created_graph["embed_id"]
+        result = dog.Embed.revoke(embed_id)
+        # Check embed is revoked and that we can't get it again
+        assert "success" in result
+        try:
```
Comment (yannmh) [inline tests/integration/api/test_api.py]: I know we us this pattern a lot, but can we switch to `assertRaises` instead (c.f. https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaises).

**Comment by yannmh** [comment]
Great work @charleslai ! Thanks these new features.

It's ready to be merged :ship:.I just made a few nitpicky comments, if you want to adress them.

File: datadog/api/graphs.py (hunk 1 above)
Comment (charleslai) [inline datadog/api/graphs.py]: Nope - removing

**Comment by charleslai** [comment]
I think I addressed all of the comments - I didn't change the `Action..` class because there is a method there called `_trigger_class_action` that I can use and this combined with `_trigger_action` seems to cover most use cases

**Comment by yannmh** [comment]
Great work, thanks @charleslai.

Let's merge it and release a new package :rocket:.

**Pull request merged**